src/mcdlutil.cpp

```diff
@@ -2569,7 +2569,7 @@ namespace OpenBabel {
               uy1=getAtom(dsNA1[i])->ry-uy;
               //coordinates of all atoms for the cycle under study are calculated}
 
-              for (j=0; j<nb; j++) {
+              for (j=0; j<nb-1; j++) {
```

Comment:
This change seems to be the one responsible for the regression reported in #1900 but without it, example 3 from the bug report (#1851), i.e. `O=C([C@@H]1C[C@H]1C1CCC1)N1C2CCC1CN(C(=O)C13CCN(CC1)C3)C2`,  gives an exception.